src/activemasternode.cpp

```diff
@@ -170,19 +170,16 @@ bool CActiveMasternode::SendMasternodePing(std::string& errorMessage)
         return false;
     }
 
-    CPubKey pubKeyMasternode;
-    CKey keyMasternode;
-
-    if (!CMessageSigner::GetKeysFromSecret(strMasterNodePrivKey, keyMasternode, pubKeyMasternode)) {
-        errorMessage = "Error upon calling GetKeysFromSecret.\n";
+    if (!privKeyMasternode.IsValid() || !pubKeyMasternode.IsValid()) {
+        errorMessage = "Error upon masternode key.\n";
         return false;
     }
```

Comment:
This should never happen, as `GetKeysFromSecret` already checks that the key is valid, when we initialize the node, and then we never change it. 
So it would be probably better to throw here.

Reply:
Yeah, i left it in this way to not make much changes on the caller side (every caller is expecting to pass the two arguments and get a boolean from the function, same as with the previous `GetKeysFromSecret`).

The best would be to return a key pair struct instead of have to pass the two ref args or well.. keep it simple and add the two getters.